jobs list

laclac102 opened the pull request:
This PR introduces the ability to:

- Fetch all pipelines executed by the current user - Retrieves a list of workflows the user has launched.
- Cancel a running pipeline – Provides a function to cancel a specific pipeline execution by ID.

<img width="1046" height="526" alt="Screenshot 2025-07-15 at 3 14 39 pm" src="https://github.com/user-attachments/assets/5da5e9d5-cd03-457b-8c89-7d8d695dff72" />

- 
Details:

- Implemented API call to list pipelines.
- Added a cancel button/action that triggers the API to stop a selected pipeline.
- Disable cancel buttons for finish or canceled pipelines

laclac102 commented on src/pages/api/listRuns.ts:
```diff
@@ -0,0 +1,33 @@
+
+const BASE_URL = process.env.NEXT_PUBLIC_SEQERA_API_URL;
+const WORKSPACE_ID = process.env.NEXT_PUBLIC_WORKSPACE_ID;
+const token = process.env.NEXT_PUBLIC_SEQERA_ACCESS_TOKEN!;
```
We had talked to Ziad and Thom several times about this. This token will be general token and used by all users in the portal, same as private key for HPC in the up coming launching feature. User will not required to type in their key or token, but their username and password to access the portal/bundle (?).

marius-mather replied:
OK. If a shared token is being used then we need to only use it on the server, not in the browser. It can't be a `NEXT_PUBLIC_` variable, but it *could* be on the NextJS server, see https://nextjs.org/docs/pages/guides/environment-variables#loading-environment-variables. If that won't work, then we may need a separate backend service that does the actual communication with the Seqera API.

laclac102 commented on src/pages/api/listRuns.ts:
```diff
@@ -0,0 +1,33 @@
+    method: "GET",
+    headers: myHeaders
+  };
+  const listRunsURL = `${BASE_URL}/workflow?workspaceId=${WORKSPACE_ID}`;
```
I had one problem with this, please lemme know if you have any idea to solve it. The Seqera API supported [pagination](https://docs.seqera.io/platform-cloud/troubleshooting_and_faqs/api_and_cli#api), but not the owner query. I'm currently get all pipelines executed on workspace -> filter by ownerId in server side -> render on client side. Then if we set the pagination for at api handler function -> the total number of pipelines executed by current user at each page will not be the same (?)

marius-mather replied:
Ah ok, if you need to filter the API results on this end then pagination is going to be tricky. The detailed API docs say the `/workflow` endpoint supports a `search` parameter: https://cloud.seqera.io/openapi/index.html#get-/workflow. But I can't find out any more information about it. If you can't see how to filter the API results then let's leave pagination for now.

laclac102 replied:
Yeah, i tried to use the search query in /workflow API to filter the ownerId but it didn't work. You can have a look on details how the output schema will be [here](tructural-bio-portal/blob/589083fac2ef58e2f0ad66f08e3c7d9b3f27301d/src/models/workflow.ts#L85)

laclac102 commented:
The latest commit includes the following updates:
- RawRunInfo updated to enhance the readability
- All public params moved to local params

marius-mather commented on src/pages/runs/runWorkflow.tsx:
```diff
@@ -102,13 +108,34 @@ export default function RunWorkflowPage() {
+        </Alert>
+      )}
+      {status === "error" && (
+        <Alert severity="error">Some errors happened!</Alert>
```
Are you able to provide any detail on the errors here? It would be good to include an error message, if it's available, so users can report problems.

laclac102 replied:
solve at [the latest commit](https://github.com/Australian-Structural-Biology-Computing/structural-bio-portal/pull/6/commits/90bd85f982ea45f2e974e7d6e39cde947c63601b)

marius-mather commented on src/pages/runs/runWorkflow.tsx:
```diff
@@ -46,7 +50,7 @@ export default function RunWorkflowPage() {
       setRunId(result);
       setStatus("done");
-      console.log("runId: ", runID);
+      setTimeout(() => router.push({ pathname: "/jobs" }), 3000);
```
Will this just automatically take them to the jobs page after a delay? Might be best to inform the user and say "taking you to the jobs page" in the alert

laclac102 replied:
Yay, agree! Solved at [this commit](https://github.com/Australian-Structural-Biology-Computing/structural-bio-portal/pull/6/commits/90bd85f982ea45f2e974e7d6e39cde947c63601b)

marius-mather commented on src/pages/runs/runWorkflow.tsx:
```diff
@@ -102,13 +108,34 @@ export default function RunWorkflowPage() {
+      alignItems="center"
+      justifyContent="center"
+    >
+      {status === "idle" && (
```
I think using the status switch is a pretty good way to show the different UI depending on the current workflow state

laclac102 replied:
solved at[ the latest commit](https://github.com/Australian-Structural-Biology-Computing/structural-bio-portal/pull/6/commits/90bd85f982ea45f2e974e7d6e39cde947c63601b)

marius-mather reviewed (changes requested):
These latest changes look good to me, just a couple of minor notes on showing feedback to the user

laclac102 commented:
Hi @marius-mather , there are some new fixes added to above commits while I'm testing the new compute Env:

- DB schema edited: removed all_in_one (the new modification of DB will be included in next PR at branch db-schema), added configProfiles and revision, which are required to execute BindFlow
- Deal with params type: current form submit all params as string -> deal with integer + booleans. 2 util functions were introduced, 1 for parse workflow schema, 1 for convert form data based on params type. All related api and controller function edited.
- Clean up form/Form provider: remove form and its onSubmit and keep custom FormProvider only.
Launcher tested and succesfully launched custom workflow with correct params type.
- Error and re-directing to job list alert added!
<img width="1232" height="619" alt="Screenshot 2025-07-21 at 1 31 06 pm" src="https://github.com/user-attachments/assets/aac75350-8980-42ee-a5ef-b9ae820ec848" />

laclac102 commented:
Hihi @marius-mather Updated: Unit test for all controllers added hihi

marius-mather commented on src/__test__/cancelWorkflow.test.ts:
```diff
@@ -0,0 +1,43 @@
+import { cancelWorkflow } from "@/controllers/cancelWorkflow";
```
These tests look great - nice and short while also testing exactly what needs to be tested

marius-mather commented on src/__test__/listRuns.test.ts:
```diff
@@ -0,0 +1,226 @@
+
+  const mockUserId = 123; // match the `ownerId: number` type
+
+  const mockRuns: RawRunInfo[] = [
```
Ouch! Did you have to type all this out? It might be worth think about how to generate this in a function in case you need to modify it for future tests, using an approach like this: https://levelup.gitconnected.com/generate-surgically-precise-test-data-3e914c46d6b4

laclac102 replied:
awesome! I will think about this. I used chatgpt to generated the above :)))

marius-mather commented on src/pages/api/cancel.ts:
```diff
@@ -1,12 +1,14 @@
-
+const BASE_URL = process.env.SEQERA_API_URL;
+const WORKSPACE_ID = process.env.WORKSPACE_ID;
+const token = process.env.SEQERA_ACCESS_TOKEN!;
```
Is this working OK only having the access token accessible on the server side? That's great if it works without further changes.

marius-mather reviewed (approved):
This is looking good from what I can see - great to see unit tests! But this PR is also getting a bit large and hard to review. Let's get this merged and do new PRs for further work.
